server/plugin/kv_test.go

```diff
@@ -145,6 +148,96 @@ func TestDeleteUser(t *testing.T) {
 	}
 }
 
+func TestGetAllUsers(t *testing.T) {
+	defer monkey.UnpatchAll()
+	ps := new(pluginStore)
+	ps.plugin = &Plugin{}
+	ps.userKV = kvstore.NewHashedKeyStore(kvstore.NewPluginStore(&plugintest.API{}), constants.UserKeyPrefix)
+	for _, test := range []struct {
+		description   string
+		setupTest     func()
+		setupAPI      func(*plugintest.API)
+		expectedError string
+	}{
+		{
+			description: "GetAllUsers: success",
+			setupTest: func() {
+				monkey.PatchInstanceMethod(reflect.TypeOf(ps), "LoadUser", func(*pluginStore, string) (*serializer.User, error) {
+					return testutils.GetSerializerUser(), nil
+				})
+			},
+			setupAPI: func(a *plugintest.API) {
+				a.On("KVList", testutils.GetMockArgumentsWithType("int", 2)...).Return(
+					[]string{
+						testutils.GetUserKey(true),
+					}, nil,
+				)
+			},
+		},
+		{
+			description: "GetAllUsers: KVList gives error",
+			setupTest:   func() {},
+			setupAPI: func(a *plugintest.API) {
+				a.On("KVList", testutils.GetMockArgumentsWithType("int", 2)...).Return(
+					nil, testutils.GetInternalServerAppError("error in loading the KVList"),
```

Comment:
KVList is a function, not an actual list. So I think this should be better please change everywhere
```suggestion
					nil, testutils.GetInternalServerAppError("error in loading the user list"),
```